src/main/java/edu/tamu/app/service/CatalogService.java

```diff
@@ -1,47 +1,115 @@
+
+        if (!catalogsFile.equals("")) {
+            ClassPathResource catalogsRaw = new ClassPathResource(catalogsFile);
+            try {
+                final JsonNode catalogsJson = objectMapper.readTree(new FileInputStream(catalogsRaw.getFile()));
+                catalogsJson.get("catalogs").fieldNames().forEachRemaining(catalogName -> {
+                    try {
+                        catalogConfigurations.put(catalogName, objectMapper.readValue(catalogsJson.get("catalogs").get(catalogName).toString(), new TypeReference<HashMap<String, Object>>() {}));
+                    } catch (IOException e) {
+                        // TODO Auto-generated catch block
+                        e.printStackTrace();
+                    }
+                });
+            } catch (JsonProcessingException e) {
+                // TODO Auto-generated catch block
+                e.printStackTrace();
+            } catch (FileNotFoundException e) {
+                // TODO Auto-generated catch block
+                e.printStackTrace();
+            } catch (IOException e) {
+                // TODO Auto-generated catch block
+                e.printStackTrace();
+            }
+        }
+    }
+
+    public List<CatalogHolding> getHoldingsByBibId(String catalogName, String bibId) {
+        Map<String,String> parameters = new HashMap<String,String>();
+        parameters.put("bibId", bibId);
+        try {
+            return objectMapper.readValue(getData("get-holdings", parameters).get("ArrayList<CatalogHolding>").toString(), new TypeReference<List<CatalogHolding>>() {});
+        } catch (IOException e) {
+            logger.error("Unable to retrieve Holdings from Catalog Service");
+            e.printStackTrace();
+            return null;
+        }
+    }
+
+    public CatalogHolding getHolding(String catalogName, String bibId, String holdingId) {
+        Map<String,String> parameters = new HashMap<String,String>();
+        parameters.put("bibId", bibId);
+        parameters.put("holdingId", holdingId);
+        try {
+            return objectMapper.readValue(getData("get-holding", parameters).get("CatalogHolding").toString(), new TypeReference<CatalogHolding>() {});
+        } catch (IOException e) {
+            logger.error("Unable to retrieve Holding from Catalog Service");
+            e.printStackTrace();
+            return null;
+        }
+    }
+
+    public Map<String,String> getCatalogConfigurationByName(String catalogName) {
+        return catalogConfigurations.get(catalogName);
+    }
+
+    private JsonNode getData(String endPoint, Map<String,String> parameters) throws IOException {
+        StringBuilder urlBuilder = new StringBuilder();
+        urlBuilder.append(catalogServiceUrl + endPoint+"/");
+        if (parameters != null && !parameters.isEmpty()) {
+            urlBuilder.append("?");
+            parameters.forEach((k,v) -> {
+                urlBuilder.append(k + "=" + v);
```

Comment:
Something seems missing to me, shouldn't there be an ampersand '&', appended between the for each.
Such as:
```suggestion
                urlBuilder.append(k + "=" + v + "&");
```

Isn't that why there is a `-1` in `urlBuilder.length()-1` below?

Reply:
Yep. Thanks for catching it.

Reply:
It's fixed and I've added the additional parameters that I left out for the calls.